sources/packages/api/src/route-controllers/program-info-request/program-info-request.controller.ts

```diff
@@ -72,4 +82,46 @@ export class ProgramInfoRequestController {
       );
     }
   }
+
+  /**
+   * Get all application of a location in a institution
+   * with Program Info Request (PIR) status completed and required
+   * @param locationId location that is completing the PIR.
+   * @returns student application list of a institution location
+   */
+  @AllowAuthorizedParty(AuthorizedParties.institution)
+  @HasLocationAccess("locationId")
+  @Get(":locationId/pir-summary")
+  async getPIRSummary(
+    @Param("locationId") locationId: number,
+    @UserToken() userToken: IInstitutionUserToken,
+  ): Promise<PIRSummaryDTO[]> {
+    const institutionDetails =
+      await this.institutionService.getInstituteByUserName(userToken.userName);
+    const requestedLoc = await this.locationService.getInstitutionLocationById(
+      locationId,
+    );
+    if (institutionDetails.id !== requestedLoc.institution.id) {
+      throw new ForbiddenException();
+    }
+    if (!institutionDetails) {
```

Comment:
Dont we need to add a exception handling before _if (institutionDetails.id !== requestedLoc.institution.id)_?

if (!institutionDetails) {
      throw new UnprocessableEntityException(
        "Not able to find a institution associated with the current user name.",
      );
    }